packages/compass-web/src/index.tsx

```diff
@@ -138,9 +193,17 @@ const CompassWeb = ({
     throw connectionError;
   }
 
-  if (connected && dataService.current) {
+  if (!connected || !dataService.current) {
```

Comment:
As promised in https://github.com/mongodb-js/compass/pull/5249#discussion_r1434215704 🙂